cmd/vcfAfs/vcfAfs.go

```diff
@@ -11,27 +11,52 @@ import (
 	"log"
 )
 
-func vcfAfs(vcfFile string, outFile string, UnPolarized bool) {
-	g, err := popgen.VcfToAfs(vcfFile, UnPolarized, false) //hardcoded DivergenceAscertainment to false
+type Settings struct {
+	UnPolarized             bool
+	PlotSelectionLikelihood string
+	LeftBound               float64
+	RightBound              float64
+	NumberOfPoints          int
+	IntegralError           float64
+	DivergenceAscertainment bool
+	D                       int //size of the ascertainment subset
+}
+
+func vcfAfs(vcfFile string, outFile string, s Settings) {
+	var err error
+	g, err := popgen.VcfToAfs(vcfFile, s.UnPolarized, s.DivergenceAscertainment)
 	exception.FatalOnErr(err)
 	f := popgen.AfsToFrequency(*g)
 	out := fileio.EasyCreate(outFile)
-	defer out.Close()
 	for i := 0; i < len(f); i++ {
-		fmt.Fprintf(out, "%f\n", f[i])
+		_, err = fmt.Fprintf(out, "%f\n", f[i])
+		exception.PanicOnErr(err)
+	}
+	err = out.Close()
```

Comment:
I was thinking, this is going to just close the out file, or does it close both that file and one vcfFile (incoming file)?

Reply:
The opening and closing of the vcfFile is handled inside the popgen.VcfToAfs function call. This just closes the output file.